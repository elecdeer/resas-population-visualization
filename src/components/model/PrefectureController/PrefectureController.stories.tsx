import { PrefectureController, PrefectureControllerProps} from "./PrefectureController";

export default {
	title: "ui/PrefectureController",
	component: PrefectureController
}

export const Default = (args: PrefectureControllerProps) => <PrefectureController {...args}/>