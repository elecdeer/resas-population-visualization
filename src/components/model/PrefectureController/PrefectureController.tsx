import React from "react";
import { CheckBox } from "../../ui/CheckBox/CheckBox";
import { CheckBoxesGridContainer } from "../../ui/CheckBoxesGridContainer/CheckBoxesGridContainer";
import { PrefecturesRes } from "../../../lib/schema/prefecturesResSchema";
import {
  PrefectureCheckStates,
  SetPrefectureCheckStates,
} from "../../../hooks/usePrefecturesState";

export type PrefectureControllerProps = {
  prefectures: PrefecturesRes["result"];
  checkState: PrefectureCheckStates;
  setCheckState: SetPrefectureCheckStates;
};

export const PrefectureController: React.FC<PrefectureControllerProps> = ({
  prefectures,
  checkState,
  setCheckState,
}) => {
  return (
    <CheckBoxesGridContainer>
      {prefectures.map((pref) => {
        const handleChangeCheck: React.ChangeEventHandler<HTMLInputElement> = (
          e
        ) => {
          console.log(`handleOnChange: ${pref.prefCode} ${e.target.checked}`);
          setCheckState(pref.prefCode, e.target.checked);
        };

        return (
          <CheckBox
            defaultChecked={checkState[pref.prefCode]}
            isChecked={checkState[pref.prefCode]}
            onChange={handleChangeCheck}
            key={pref.prefCode}
          >
            {pref.prefName}
          </CheckBox>
        );
      })}
    </CheckBoxesGridContainer>
  );
};
