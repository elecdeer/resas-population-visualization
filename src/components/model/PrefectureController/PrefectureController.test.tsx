import { PrefectureController } from "./PrefectureController";
import { render } from "@testing-library/react";

const mockPrefData = [
  {
    prefCode: 1,
    prefName: "北海道",
  },
  {
    prefCode: 2,
    prefName: "青森県",
  },
  {
    prefCode: 3,
    prefName: "岩手県",
  },
];

const mockStateData = {
  1: true,
  2: false,
  3: false,
};

test("PrefectureController Component Snapshot Test", () => {
  const { asFragment } = render(
    <PrefectureController
      prefectures={mockPrefData}
      checkState={mockStateData}
      setCheckState={(key, checked) => {}}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
