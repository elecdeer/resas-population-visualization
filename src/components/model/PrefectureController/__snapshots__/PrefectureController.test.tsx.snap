// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrefectureController Component Snapshot Test 1`] = `
<DocumentFragment>
  <div
    class="check-boxes-grid-container"
  >
    <label
      class="checkbox"
    >
      <input
        checked=""
        class="input"
        type="checkbox"
      />
      <span
        class="label"
      >
        北海道
      </span>
    </label>
    <label
      class="checkbox"
    >
      <input
        class="input"
        type="checkbox"
      />
      <span
        class="label"
      >
        青森県
      </span>
    </label>
    <label
      class="checkbox"
    >
      <input
        class="input"
        type="checkbox"
      />
      <span
        class="label"
      >
        岩手県
      </span>
    </label>
  </div>
</DocumentFragment>
`;
